File: LivrusVirtus/VCPP6/FoxToolKit/LibFireBird/ConAssuntos.cpp
#include "resource.h"
#include "ConAssuntos.h"
#include "Principal.h"
#include "ConsultasSQL.h"

CfrmConAssuntos *frmConAssuntos = (CfrmConAssuntos *) NULL;

FXDEFMAP(CfrmConAssuntos) CfrmConAssuntosMap[]={
  FXMAPFUNC(SEL_COMMAND, CfrmConAssuntos::ID_FRMCONSULTAS_BTNPESQUISAR, 
    CfrmConAssuntos::onBtnPesquisar),
  FXMAPFUNC(SEL_COMMAND, CfrmConAssuntos::ID_FRMCONSULTAS_BTNLIMPAR, 
    CfrmConAssuntos::onBtnLimpar),
  FXMAPFUNC(SEL_SELECTED,CfrmConAssuntos::ID_FRMCONSULTAS_GRIDCONSULTA,
    CfrmConAssuntos::onGridSelected),
  FXMAPFUNC(SEL_CHANGED,CfrmConAssuntos::ID_FRMCONSULTAS_GRIDCONSULTA,
    CfrmConAssuntos::onGridChanged)
};
FXIMPLEMENT(CfrmConAssuntos,CfrmConsultas,CfrmConAssuntosMap,
  ARRAYNUMBER(CfrmConAssuntosMap))

CfrmConAssuntos::CfrmConAssuntos(FXWindow* owner):
  CfrmConsultas(owner, STR_CONASSUNTO) {
  IniciaControles();
}

void CfrmConAssuntos::IniciaControles() {
  this->labDescricao->setText("Assunto:");
  this->gridConsulta->setColumnText(1, "Assunto");
}

void CfrmConAssuntos::create(){
  FXDialogBox::create();
  show(PLACEMENT_SCREEN);
}

CfrmConAssuntos::~CfrmConAssuntos(){  
}

long CfrmConAssuntos::onBtnPesquisar(FXObject*,FXSelector,void*) {
  PesquisarDados(
    ConsSQL->Assunto('S',"", edtDescricao->getText()), STR_ASSNENC, 2);
  return true;
}

long CfrmConAssuntos::onBtnLimpar(FXObject*,FXSelector,void*) {
  LimparDados();
  return true;
}

long CfrmConAssuntos::onGridSelected(FXObject*,FXSelector,void* ptr) {
  FXTableRange *escala = (FXTableRange*) ptr;
  setLabRegistros(escala->fm.row);
  return true;
}

long CfrmConAssuntos::onGridChanged(FXObject*,FXSelector,void* ptr) {
  FXTablePos *pos = (FXTablePos*) ptr;
  setLabRegistros(pos->row);
  return true;
}

File: LivrusVirtus/FoxToolKit/FXPython/ConAssuntos.py
from FXPy.fox import *
from Consultas import frmConsultas
from resource import Resource
from ConsultasSQL import ConsultasSQL
import Principal

class frmConAssuntos(frmConsultas):

  def __init__(self, owner):
    frmConsultas.__init__(self, owner, Resource.STR_CONASSUNTO)  

    FXMAPFUNC(self, SEL_COMMAND, frmConAssuntos.ID_FRMCONSULTAS_BTNPESQUISAR, 
      frmConAssuntos.onBtnPesquisar)
    FXMAPFUNC(self, SEL_COMMAND, frmConAssuntos.ID_FRMCONSULTAS_BTNLIMPAR, 
      frmConAssuntos.onBtnLimpar)
    FXMAPFUNC(self, SEL_SELECTED, frmConAssuntos.ID_FRMCONSULTAS_GRIDCONSULTA, 
      frmConAssuntos.onGridSelected)
    FXMAPFUNC(self, SEL_CHANGED, frmConAssuntos.ID_FRMCONSULTAS_GRIDCONSULTA, 
      frmConAssuntos.onGridChanged)
    
    self.labDescricao.setText("Assunto:")
    self.gridConsulta.setItemText(0, 2, "Assunto")

  def create(self):
    frmConsultas.create(self)
  
  def onBtnPesquisar(self, sender, sel, ptr):
    self.PesquisarDados(ConsultasSQL.ConsSQL.Assunto('S',"", 
      self.edtDescricao.getText()), Resource.STR_ASSNENC, 2)
    return True
  
  def onBtnLimpar(self, sender, sel, ptr):
    self.LimparDados()
    return True
  
  def onGridSelected(self, sender, sel, ptr):
    self.setLabRegistros(sender.getCurrentRow())
    return True
  
  def onGridChanged(self, sender, sel, ptr):
    self.setLabRegistros(sender.getCurrentRow())
    return True
